### PR TITLE
Fixing async Task Execution

### PR DESCRIPTION
**Pull request by Alea81**

### CODE REVIEW

File: src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs [hunk 1]
```diff
@@ -145,7 +147,7 @@ public async Task Execute(ExecuterSettings arguments)
         /// <param name="policy">Policy</param>
         protected abstract void DeletePolicy(PolicyHttpClient policyClient, Guid projectId, PolicyConfiguration policy);
 
-        private void HandleChanges(
+        private async void HandleChanges(
```
Comment (pascalberger) [inline src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs]: Should return Task

File: src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs [hunk 2]
```diff
@@ -219,7 +218,7 @@ public async Task Execute(ExecuterSettings arguments)
                             }
                             else
                             {
-                                resultList.Add(() => this.UpdatePolicy(policyClient, types, projectId, repository, currentPolicy, policy, serverPolicy));
+                                resultList.Add(Task.Run(() => this.UpdatePolicy(policyClient, types, projectId, repository, currentPolicy, policy, serverPolicy)));
```
Comment (pascalberger) [inline src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs]: Why calling `Task.Run` here?

File: src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs [hunk 3]
```diff
@@ -229,7 +228,7 @@ public async Task Execute(ExecuterSettings arguments)
 
                 if (!hasMatch)
                 {
-                    resultList.Add(() => this.CreatePolicy(policyClient, types, projectId, repository, currentPolicy, policy));
+                    resultList.Add(Task.Run(() => this.CreatePolicy(policyClient, types, projectId, repository, currentPolicy, policy)));
```
Comment (pascalberger) [inline src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs]: Why calling `Task.Run` here?

Comment (Alea81) [reply src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs]: Did not find a better solution to await void methods:
https://stackoverflow.com/questions/13636648/wait-for-a-void-async-method

File: src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs (hunk 2 above)
Comment (Alea81) [reply src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs]: Did not find a better solution to await void methods:
https://stackoverflow.com/questions/13636648/wait-for-a-void-async-method

File: src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs (hunk 3 above)
Comment (Alea81) [reply src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs]: Found a better way

File: src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs (hunk 2 above)
Comment (Alea81) [reply src/AzureDevOpsPolicyConfigurator/Logic/PolicyExecuterBase.cs]: Found a better way

**Pull request merged**